text/0000-publish-operation.md

```diff
@@ -0,0 +1,144 @@
+# Meta
+[meta]: #meta
+- Name: Publish Operation
+- Start Date: 2022-02-22
+- Author(s): Juan Bustamante (@jbustamante)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Split the process of creating and saving the Image in two different operations.
+- Export: will create the Image and saves it to disk in [OCI Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format. When the `daemon` flag is enabled it could save all the metadata that can't be saved into the daemon in a separate report.
+- Publish: will take the image generated by Export and push it into the Daemon or in the Registry.
+
+# Definitions
+[definitions]: #definitions
+
+- A [Platform](https://buildpacks.io/docs/concepts/components/platform/) uses a lifecycle, Buildpacks (packaged in a builder), and application source code to produce an OCI image.
+- A [Lifecycle](https://buildpacks.io/docs/concepts/components/lifecycle/) orchestrates Buildpacks execution, then assembles the resulting artifacts into a final app image.
+- A Daemon is a service, popularized by Docker, for downloading container images, and executing and managing containers from those images.
+- A Registry is a long-running service used for storing and retrieving container images.
+- An [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) is the directory structure for OCI content-addressable blobs and location-addressable references.
+- An [image index](https://github.com/opencontainers/image-spec/blob/main/image-index.md) is a higher-level manifest which points to a list of manifests and descriptors.
+- An [Image manifest](https://github.com/opencontainers/image-spec/blob/main/manifest.md) provides a configuration and set of layers for a single container image for a specific architecture and operating system.
+- A [config](https://github.com/opencontainers/image-spec/blob/main/descriptor.md) is a property references a configuration object for a container, by digest. It must support the following media type `application/vnd.oci.image.config.v1+json`
+
+# Motivation
+[motivation]: #motivation
+
+Because the [Image.Digest](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/v1#Image) method from the GCR library would compute the Digest value before pushing it to a Registry, it allows us to create the image and add  metadata require like annotations or even sign it and save it to disk and be sure that when the Image is push to the registry the Digest will not change. (See the [thread](https://cloud-native.slack.com/archives/C033DV9EBDF/p1644523524402149) on Slack).
```

Comment:
I'm not understanding the motivation here. What's the problem that this is trying to solve?

Reply:
Hi @jromero , I rewrote the whole idea, I hope is more clear now